### PR TITLE
Add endpoint to get questions by level and module

### PR DESCRIPTION
**Pull request by SemyonKorostik**

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/controllers/QuestionController.java [hunk 1]
```diff
@@ -34,6 +37,22 @@ public QuestionDTO getQuestion(@PathVariable("id") Long id) {
...
+                .collect(Collectors.toList());
+    }
+
+    @ApiOperation(value = "Get questions from the database by it's module id")
+    @GetMapping("/module/{moduleId}")
+    public List<QuestionDTO> getQuestionsByModule(@PathVariable("moduleId") Long moduleId) {
+        return questionService.getQuestionsByModuleId(moduleId).stream()
+                .map(questionConverter::convertToDTO)
+                .collect(Collectors.toList());
+    }
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/QuestionController.java:62]: please, ask a frontend team here, but I think they need a single endpoint for getting all questions by both, test and module

File: src/main/java/com/team4/testingsystem/controllers/QuestionController.java [hunk 2]
```diff
@@ -34,6 +37,22 @@ public QuestionDTO getQuestion(@PathVariable("id") Long id) {
...
+
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/QuestionController.java:63]: Don't forget that the expected result for listening is quite different. I would suggest creating a special endpoint for it

File: src/main/java/com/team4/testingsystem/controllers/QuestionController.java [hunk 3]
```diff
@@ -34,6 +37,22 @@ public QuestionDTO getQuestion(@PathVariable("id") Long id) {
         return questionConverter.convertToDTO(questionService.getById(id));
     }
 
+    @ApiOperation(value = "Get questions from the database by it's level id")
+    @GetMapping("/level/{levelId}")
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/QuestionController.java]: don't think frontend needs to know about level and module id. Let's just get the value and map it to corresponding enum

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

File: src/main/java/com/team4/testingsystem/controllers/QuestionController.java (hunk 1 above)
Comment (maxim12321) [reply src/main/java/com/team4/testingsystem/controllers/QuestionController.java:62]: Yep, they do need endpoint with 2 parameters: level **and** module

File: src/test/java/com/team4/testingsystem/controllers/QuestionControllerTest.java [hunk 4]
```diff
@@ -47,6 +46,16 @@ void getQuestion() {
...
     }
 
+    @Test
+    void getQuestionsByLevelAndModuleName() {
+        List<Question> questions = new ArrayList<>();
+        questions.add(EntityCreatorUtil.createQuestion());
+        Mockito.when(questionService.getQuestionsByLevelAndModuleName(any(), any())).thenReturn(questions);
+        Assertions.assertEquals(questions.stream()
+                .map(questionConverter::convertToDTO)
+                .collect(Collectors.toList()),questionController.getQuestions(any(), any()));
```
Comment (maxim12321) [inline src/test/java/com/team4/testingsystem/controllers/QuestionControllerTest.java]:
```
List<QuestionDTO> expectedQuestions = questions.stream()
        .map(questionConverter::convertToDTO)
        .collect(Collectors.toList())
Assertations.assertEquals(expectedQuestions, questionController.getQuestions(any(), any()));
```

File: src/main/java/com/team4/testingsystem/controllers/QuestionController.java [hunk 5]
```diff
@@ -34,6 +38,15 @@ public QuestionDTO getQuestion(@PathVariable("id") Long id) {
         return questionConverter.convertToDTO(questionService.getById(id));
     }
 
+    @ApiOperation(value = "Get questions from the database by it's level and module")
+    @GetMapping("/")
+    public List<QuestionDTO> getQuestions(@RequestParam("level") String level,
+                                          @RequestParam("module") String module) {
```
Comment (maxim12321) [inline src/main/java/com/team4/testingsystem/controllers/QuestionController.java]: Why don't you use `Levels` and `Modules` enums?

File: src/test/java/com/team4/testingsystem/controllers/QuestionControllerTest.java [hunk 6]
```diff
@@ -47,6 +46,16 @@ void getQuestion() {
         Assertions.assertEquals(questionDTO, result);
     }
 
+    @Test
+    void getQuestionsByLevelAndModuleName() {
+        List<Question> questions = new ArrayList<>();
+        questions.add(EntityCreatorUtil.createQuestion());
```
Comment (maxim12321) [inline src/test/java/com/team4/testingsystem/controllers/QuestionControllerTest.java]:
```
List<Question> questions = Lists.list(EntityCreatorUtil.createQuestion())
```

**Review by maxim12321 — CHANGES_REQUESTED** [review]

**Review by maxim12321 — APPROVED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

**Pull request merged**